File: app/src/main/java/org/rm3l/router_companion/tiles/status/wireless/filter/impl/ShowOnlyHostsWithWANAccessDisabledFilterVisitorImpl.java
package org.rm3l.router_companion.tiles.status.wireless.filter.impl;

import android.support.annotation.NonNull;
import com.google.common.base.Predicate;
import com.google.common.collect.Sets;
import java.util.Set;
import org.rm3l.router_companion.resources.Device;
import org.rm3l.router_companion.tiles.status.wireless.filter.ClientsFilterVisitor;

public class ShowOnlyHostsWithWANAccessDisabledFilterVisitorImpl implements ClientsFilterVisitor {

    private boolean mShowOnlyWanAccessDisabledHosts;

    public ShowOnlyHostsWithWANAccessDisabledFilterVisitorImpl(
            boolean showOnlyWanAccessDisabledHosts) {
        this.mShowOnlyWanAccessDisabledHosts = showOnlyWanAccessDisabledHosts;
    }

    @NonNull
    @Override
    public Set<Device> visit(@NonNull Set<Device> devices) {
        return Sets.filter(devices, new Predicate<Device>() {
            @Override
            public boolean apply(Device device) {

                return (!mShowOnlyWanAccessDisabledHosts || (device.getWanAccessState()
                        == Device.WANAccessState.WAN_ACCESS_DISABLED));
            }
        });
    }
}

File: app/src/main/java/org/rm3l/router_companion/tiles/status/wireless/filter/impl/ShowOnlyHostsWithWANAccessDisabledFilterVisitorImpl.kt
package org.rm3l.router_companion.tiles.status.wireless.filter.impl

import org.rm3l.router_companion.resources.Device
import org.rm3l.router_companion.resources.Device.WANAccessState.WAN_ACCESS_DISABLED
import org.rm3l.router_companion.tiles.status.wireless.filter.ClientsFilterVisitor

class ShowOnlyHostsWithWANAccessDisabledFilterVisitorImpl(private val mShowOnlyWanAccessDisabledHosts: Boolean) :
    ClientsFilterVisitor {

    override fun visit(devices: Set<Device>) =
        devices.filter {
            !mShowOnlyWanAccessDisabledHosts || (it.wanAccessState
                    == WAN_ACCESS_DISABLED)
        }.toSet()
}
